### DISCUSSION TITLE
Run Oqtane in Kubernetes Container

### DISCUSSION THREAD

**Discussion by milgro-ict** [oqtane/oqtane.framework#4649] [Q&A]
We have been running our DNN asp.net application for many years and we are changing our architecture to Blazor.
We already have converted some modules to a blazor app and each run seperately in a Kuberneter container. This does mean that we do not have the DNN menu and user control included in those apps.
If we want to use oqtane iso DNN, how can we best include the blazor apps in Oqtane?
From the documentation I can read that we have to create nuget packages from the apps and include them in Oqtane.
Does this mean that both Oqtane and the apps all run in the same container?
Or is there a way this can be split?
I hope my question is clear enough and hope to get an answer from you soon.

**Comment by sbwalker** [comment]
It sounds like you want to build a system with a micro-service architecture. Oqtane (similar to DNN) is based on a modular monolithic architecture... which is summarized in Oqtane's tagline "Build Applications, Not Infrastructure". The front-end module components are developed and deployed independently - but physically they will all reside in the same container. However each module component can call back-end services which exist in separate containers, if that is what you are trying to achieve. Essentially I am referring to a micro-frontend architecture: https://www.oqtane.org/blog/!/47/blazor-and-micro-frontend-applications-mfe

**Reply by milgro-ict** [reply]
Thank you for your answer. 
Your blog is very usefull in determining our strategy for the new architecture.
